daffodil-tdml-processor/src/test/scala/org/apache/daffodil/tdml/TestTDMLRunner2.scala

```diff
@@ -424,11 +424,26 @@ abc # a comment
   }
 
   @Test def testTDMLWithInvalidDFDLSchemaEmbedded() = {
+    val testSuite =
+      <tdml:testSuite suiteName="theSuiteName" xmlns:tns={ tns } xmlns:tdml={ tdml } xmlns:dfdl={ dfdl } xmlns:xs={ xsd }>
+        <!-- This embedded schema has validation errors. -->
+        <tdml:defineSchema name="mySchema">
+          <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+          <xs:format ref="tns:GeneralFormat" />
+          <xs:element name="data" type="fn:string" dfdl:lengthKind="delimited" />
+        </tdml:defineSchema>
+        <tdml:parserTestCase name="test1" root="data" model="mySchema">
+          <tdml:document>
+            <tdml:documentPart type="text"><![CDATA[abcdef]]></tdml:documentPart>
+          </tdml:document>
+          <tdml:errors>
+            <tdml:error>Cannot resolve 'fn:string' as a QName</tdml:error>
+          </tdml:errors>
+        </tdml:parserTestCase>
+      </tdml:testSuite>
 
-    val testDir = ""
-    val aa = testDir + "tdml-with-embedded-schema-errors.tdml"
-    lazy val runner = new DFDLTestSuite(Misc.getRequiredResource(aa), validateTDMLFile = false, validateDFDLSchemas = true)
-    runner.runOneTest("test1")
+    lazy val ts = new DFDLTestSuite(testSuite, validateTDMLFile = false, validateDFDLSchemas = true)
```

Comment:
I'm passing a scala.xml.Elem as the first argument.  The factory object Runner has an apply method taking a scala.xml.Elem, but it doesn't have any extra parameters like the apply method taking a dir and file.  May I add these extra parameters to the scala.xml.Elem apply method so that I can call Runner instead of new DFDLTestSuite with the arguments (testSuite, validateTDMLFile = false, validateDFDLSchemas = true)?

Reply:
Also, do we really need "lazy val ts =" in methods when the val is used immediately afterwards?  I think we can replace "lazy val ts =" with "val ts =" in these methods without changing what these methods are doing.

Reply:
IntelliJ didn't warn me, but when I ran sbt compile, I found out you can't add default arguments to two overloaded apply methods:

`in object Runner, multiple overloaded alternatives of method apply define default arguments`

I will work around by adding a new apply method that passes both scala.xml.Elem and validateDFDLSchema as required arguments.

Reply:
Sounds right.

Reply:
lazy val valName = .... 
Is seldom needed in method code. There are a few places where we define two structures that refer to each other where we need this, but yeah, generally in method code val suffices.